perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1123,6 +1218,64 @@ void MapBasedPredictionNode::objectsCallback(const TrackedObjects::ConstSharedPt
     "debug/processing_time_ms", processing_time_ms);
 }
 
+void MapBasedPredictionNode::updateCrosswalkUserHistory(
+  const std_msgs::msg::Header & header, const TrackedObject & object, const std::string & object_id)
+{
+  CrosswalkUserData crosswalk_user_data;
+  crosswalk_user_data.header = header;
+  crosswalk_user_data.tracked_object = object;
+  if (crosswalk_users_history_.count(object_id) == 0) {
+    crosswalk_users_history_.emplace(object_id, std::deque<CrosswalkUserData>{crosswalk_user_data});
+  } else {
+    crosswalk_users_history_.at(object_id).push_back(crosswalk_user_data);
+  }
+}
+
```

Comment:
```suggestion
  const CrosswalkUserData crosswalk_user_data = [&](){
  CrosswalkUserData crosswalk_user_data;
  crosswalk_user_data.header = header;
  crosswalk_user_data.tracked_object = object;
  if (crosswalk_users_history_.count(object_id) == 0) {
    crosswalk_users_history_.emplace(object_id, std::deque<CrosswalkUserData>{crosswalk_user_data});
    return  crosswalk_user_data;
  } 
    crosswalk_users_history_.at(object_id).push_back(crosswalk_user_data);
  return CrosswalkUserData crosswalk_user_data;
}

```

Reply:
Could you elaborate why this change please? does not seem to be equivalent to the original code...

Reply:
Sorry, I got confused with the intention of this function, The current implementation is fine, but I recommend inserting an early return on the if condition and getting rid of the else and its brackets, they make the code harder to read.

Reply:
no problem. I reduced the conditional here by removing else statement